jina/executors/frameworks.py

```diff
@@ -25,11 +25,13 @@ def device(self):
             please use the environment variable `CUDA_VISIBLE_DEVICES`.
         """
         try:
+            import GPUtil
+            self.on_gpu = True if len(GPUtil.getGPUs()) > 0 else False
```

Comment:
or change this line like this
```
if self.on_gpu:
     self.on_gpu = True if len(GPUtil.getGPUs()) > 0 else False
```

Reply:
1. `self.on_gpu = len(GPUtil.getGPUs())`
2. kinda waste to use a third party package just for detecting if GPU exist

Reply:
1. now there is no a better way to detect if GPU exist, so I use `GPUtil` to detect.
2. sorry about that, at the first commit, I tested in my GPU machine, I ignored this part. now I test in my MAC, it works
```
 def encode(self, data: 'np.ndarray', *args, **kwargs) -> 'np.ndarray':
        """

        :param data: a 1d array of string type in size `B`
        :return: an ndarray in size `B x D`
        """
        token_ids_batch = []
        mask_ids_batch = []
        for c_idx in range(data.shape[0]):
            token_ids = self.tokenizer.encode(
                data[c_idx], pad_to_max_length=True, max_length=self.max_length)
            mask_ids = [0 if t == self.tokenizer.pad_token_id else 1 for t in token_ids]
            token_ids_batch.append(token_ids)
            mask_ids_batch.append(mask_ids)
        token_ids_batch = self.array2tensor(token_ids_batch)
        mask_ids_batch = self.array2tensor(mask_ids_batch)
```
it will use `on_gpu` before set device.